sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provide assert macros to use as pre-conditions.
+ *
+ * @remark Asserts are turned ON even if `NDEBUG` is defined.
+ *
+ */
+
+#pragma once
+
+#if defined(NDEBUG)
+#define AZURE_SDK_INTERNAL_IMPLEMENTATION_OVERRIDE_NDEBUG
+#undef NDEBUG
+#endif
+
+#include <cassert>
+#include <stdexcept>
+
+#define AZURE_ASSERT(exp) assert(exp)
```

Comment:
I recommend wrapping `exp` in another set of parentheses here so `AZURE_ASSERT(some_template_variable<T, U>)` looks like `assert((some_template_variable<T, U>))` to the preprocesser (a single argument to `assert`) instead of the ill-formed `assert(some_template_variable<T, U>)`  (two arguments `some_template_variable<T` and `U>`).